api/src/backend/api/models.py

```diff
@@ -165,6 +172,281 @@ class JSONAPIMeta:
         resource_name = "users"
 
 
+class APIKeyUser:
+    """
+    A user-like object specifically for API key authentication.
+
+    This class provides a proper user identity for API key requests instead of using
+    AnonymousUser, which improves security and clarity in the authentication system.
+    """
+
+    def __init__(self, api_key_id, api_key_name, tenant_id, role=None):
+        self.api_key_id = api_key_id
+        self.api_key_name = api_key_name
+        self.tenant_id = tenant_id
+        self.role = role  # Store the role for RBAC
+        # Set a synthetic ID for consistency with User interface
+        self.id = f"api_key_{api_key_id}"
+        self.pk = self.id
+
+    @property
+    def is_authenticated(self):
+        """API key users are always considered authenticated."""
+        return True
+
+    @property
+    def is_anonymous(self):
+        """API key users are not anonymous."""
+        return False
+
+    @property
+    def is_active(self):
+        """API key users are considered active by default."""
+        return True
+
+    def __str__(self):
+        return f"APIKeyUser({self.api_key_name})"
+
+    def __repr__(self):
+        return f"APIKeyUser(api_key_id={self.api_key_id}, api_key_name={self.api_key_name}, tenant_id={self.tenant_id}, role={self.role})"
+
+
+class APIKeyManager(APIKeyManager):
+    """
+    Custom manager for APIKey that handles Row Level Security and tenant context.
+    """
+
+    def get_queryset(self):
+        """Override to ensure RLS is properly applied."""
+        return super().get_queryset()
+
+    def get_usable_keys(self, tenant_id=None):
+        """
+        Get all usable (non-revoked, non-expired) API keys.
+        If tenant_id is provided, filter by tenant.
+        """
+        queryset = self.filter(revoked=False)
+        if tenant_id:
+            queryset = queryset.filter(tenant_id=tenant_id)
+        return queryset
+
+    def get_from_key(self, key: str, tenant_id=None):
+        """
+        Get API key from the raw key string.
+        Uses the prefix for efficient lookup while respecting tenant boundaries.
+        """
+        try:
+            prefix, _, _ = key.partition(".")
+            if not prefix:
+                raise self.model.DoesNotExist("Invalid key format")
+
+            # Use all keys, not just usable ones, so we can give proper error messages for revoked/expired keys
+            queryset = self.all()
```

Comment:
I believe this doesn't work, we have RLS so if the tenant is not fixed for the session RLS will block the select, maybe we need to use `self.using(MainRouter.admin_db)`. I don't know if we really want this